File: outras_linguagens/Swift/TempConv/Sources/TempConv/TempConv.swift
//
//  TempConv.swift
//  
//
//  Created by Paulo Vieira on 22/07/22.
//

import Foundation

public struct TempConv {
    public static func celsiusToFahrenheit(_ temperature: Temperature) -> Temperature {
        return Temperature(
            value: temperature.value * 9 / 5 + 32,
            scale: .fahrenheit
        )
    }
    
    public static func fahrenheitToCelsius(_ temperature: Temperature) -> Temperature {
        return Temperature(
            value: (temperature.value - 32) * 5 / 9,
            scale: .celsius
        )
    }
}

File: typescript/src/tempconv.ts
import { Celsius } from './scales/Celsius';
import { Fahrenheit } from './scales/Fahrenheit';

export const ABSOLUTE_ZERO_C = new Celsius(-273.15);

export const FREEZING_C = new Celsius(0);

export const BOILING_C = new Celsius(100);

export const celsius_to_fahrenheit = (c: number) => new Fahrenheit((c * 9) / 5 + 32);

export const fahrenheit_to_celsius = (f: number) => new Celsius(((f - 32) * 5) / 9);
